coderd/database/migrations/000274_workspace_modules.up.sql

```diff
@@ -0,0 +1,14 @@
+ALTER TABLE
```

Comment:
I'm curious why you opted for a new table rather than using `workspace_resources`?

Reply:
Given that you can have multiple versions of a module, having all that data in `workspace_resources` would likely result in a lot of excess data storage.

Reply:
I don't see how having it in this separate table would improve that?

Reply:
Yeah I looked at the proposed schema more closely and noticed that each module is linked to a specific provisioner job.

I think there is an argument to be made that each version of a given module should only be stored once in the database. That same module version can then be referenced by multiple workspace resources presumably at a different module path?

Reply:
We don't store all terraform resources in the `workspace_resources` table. We [filter out](https://github.com/coder/coder/blob/ea2b6b7cbc9022ec92ee49f74c75aefd77613360/provisioner/terraform/resources.go#L594) everything Coder related. So to track modules like [personalize](https://registry.coder.com/modules/personalize) which only create a `coder_script` resource, we can't rely on just the resources table.